localstack/config.py

```diff
@@ -245,11 +245,28 @@ def ping(host):
 
 
 def in_docker():
-    """ Returns True if running in a docker container, else False """
+    """
+    Returns True if running in a docker container, else False
+    Ref. https://docs.docker.com/config/containers/runmetrics/#control-groups
+    """
     if not os.path.exists('/proc/1/cgroup'):
         return False
+    try:
+        if any([
+            os.path.exists('/sys/fs/cgroup/memory/docker/'),
+            any(['docker-' in file_names for file_names in os.listdir('/sys/fs/cgroup/memory/system.slice')]),
+            os.path.exists('/sys/fs/cgroup/docker/'),
+            any(['docker-' in file_names for file_names in os.listdir('/sys/fs/cgroup/system.slice/')]),
+        ]):
+            return False
+    except FileNotFoundError:
+        pass
     with open('/proc/1/cgroup', 'rt') as ifh:
-        return 'docker' in ifh.read()
+        os_hostname = open('/etc/hostname', 'rt').read().strip()
+        lines_list = ifh.read().split('\n')
+        if any([os_hostname in element for element in lines_list]) or 'docker' in ifh.read():
```

Comment:
nitpick: Technically, `split` is not required here, as we're checking the content of all lines. Could be simplified to:
```

        os_hostname = open('/etc/hostname', 'rt').read().strip()
        content = ifh.read()
        if os_hostname in content or 'docker' in ifh.read():
            return True
```